### DISCUSSION TITLE
Logical OR in filter with sdk-js?

### DISCUSSION THREAD

**Discussion by sagerio** [directus/directus#4531] [Archive]
Hi all,

I'm very excited about this framework - thanks a lot! - and currently porting a small project as a POC, using sdk-js version 9.0.0-rc.49 and Typescript.

Now I'm struggeling with the task to use a logical OR in the filter of a query:

Using the following works fine, but is an AND shortcut:

```
const target = await this.directus.items("users").read({  
  filter: {  
    userId: {
      _eq: 1
    },
    provider: {
      _eq: "google.com"
    }
  }
});
```

Using the way described in the docs:

```
const target = await this.directus.items("users").read({  
  filter: {
    _or: [
      {
        userId: {
          _eq: 1
        },
      },
      {
        provider: {
          _eq: "google.com"
        }
      }
    ]
  }
});
```

gives the error:

> No overload matches this call. The last overload caused the following error:  
> The type "{ _or: ({ uid: { _eq: string; }; } | { provider: { _eq: string; }; })[]; }" cannot be assigned to type  
>"{ \<S extends PrimaryKey>(predicate: (value: PrimaryKey, index: number, array: PrimaryKey[]) => value is S, thisArg?: any): S[]; (predicate: (value: PrimaryKey, index: number, array: PrimaryKey[]) => unknown, thisArg?: any): PrimaryKey[]; }".  

> "_or" is not available in type "{ \<S extends PrimaryKey>(predicate: (value: PrimaryKey, index: number, array: PrimaryKey[]) => value is S, thisArg?: any): S[]; (predicate: (value: PrimaryKey, index: number, array: PrimaryKey[]) => unknown, thisArg?: any): PrimaryKey[]; }"

What I'm doing wrong?

Thanks a lot,
Robert

**Comment by rijkvanzanten** [comment]
Heya! That structure looks good. The TypeScript error you’re getting is caused by some inconsistencies in the types of the SDK now. This is already being worked on in #4510. In the meantime, you can work around it by tacking an "as any" at the end of that filter to tell TS to hush

**Reply by sagerio** [reply]
Works!
Thank you so much! :thumbsup:

**Reply by davidsandoz** [reply]
@rijkvanzanten I see that #4510 got merged, but I still get that error. Is this now being worked in another PR?

**Reply by rijkvanzanten** [reply]
Not to my knowledge. If there's a problem in the SDK, please open [a new issue](https://github.com/directus/directus/issues/new?assignees=&labels=Bug+%28Potential%29&template=bug_report.yml) 👍🏻

**Reply by davidsandoz** [reply]
@rijkvanzanten Done, I've opened #7475.
